client/app/stores/account_store.coffee

```diff
@@ -268,15 +287,34 @@ class AccountStore extends Store
 
         return mb
 
-    getError: -> return _newAccountError
 
-    isWaiting: -> return _newAccountWaiting
+    getError: ->
+        return _newAccountError
+
+
+    isWaiting: ->
+        return _newAccountWaiting
+
 
     isMailboxRefreshing: (mailboxID)->
         _mailboxRefreshing[mailboxID] > 0
 
+
     getMailboxRefresh: (mailboxID) ->
         if _mailboxRefreshing[mailboxID] > 0 then 0.9 else 0
 
+    # Returns corresponding mailbox for given message and account.
+    getMailbox: (message, account) ->
+        for boxID of message.mailboxIds when boxID in account.favorites
```

Comment:
either message is a Immutable.Map and you need to use .get(mailboxIds) or it's not and you need to use direct call. You mix both in this function.